api/api/services/internal_tasks/internal_tasks_service.py

```diff
@@ -741,7 +756,11 @@ async def _fetch_previous_task_inputs(
         )
 
         validated = list[dict[str, Any]]()
-        async for run in self.storage.task_runs.fetch_task_run_resources(task.task_uid, previous_run_query):
+        async for run in self.storage.task_runs.fetch_task_run_resources(
```

Comment:
@guillaq this still use the user's tenant. Could you please explain what you had in mind to circumvent this ? Thank you!

Reply:
are we forced to use a WORKFLOW_AI_TENANT_UID and use a special storage for this case ?

Reply:
or use an agent_input collection.